src/core/utils/helpers/lazy-load.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react";
+import { useIntersection } from "react-use";
+
+export const useListItemLazyLoad = () => {
+  const listItemRef = useRef(null);
+  const intersection = useIntersection(listItemRef, {
+    root: null,
+    rootMargin: "0%",
+    threshold: 0.33
+  });
+  const isInViewPort = Boolean(intersection?.isIntersecting);
+  const [hasBeenVisible, setHasBeenVisible] = useState<boolean | null>(null);
+
+  // We need to track if the item has been visible already.
+  // In that way we can make it stay visible when scrolling back up.
+  useEffect(() => {
+    if (hasBeenVisible || hasBeenVisible !== null) {
+      return;
+    }
```

Comment:
Consider if we can simplify `hasBeenVisible` to a boolean with an initial value of `false`.

To me it seems as if we never set til to `false` anyway and I cannot see why we would need to distinguish between the two values anyway.

Reply:
Yes indeed. Brain fart removed.